storage/innobase/log/log0crypt.cc

```diff
@@ -240,6 +245,8 @@ bool log_crypt_init()
 
   info.key_version= 0;
 func_exit:
+  tmp_key_version= info.key_version;
+  memcpy(tmp_crypt_key, info.crypt_key, MY_AES_BLOCK_SIZE);
```

Comment:
Is there any way to reduce the risk of leaking the encryption key? As far as I understand, here we would copy the key even if key rotation is not supported.

Could we fetch the old key version on demand? Can we store the used key version somewhere within the encrypted block? If not, could we at least perform some additional steps to ensure that the `tmp_crypt_key` will not be written to a swap file (`mlock`) or core dump files (`madvise`)?

Based on the description, there might also be a problem with persistent encrypted tables when the key is rotated and a previously evicted page needs to be read in. How would this case work?